src/Bicep.Core/Syntax/ImportSpecification.cs

```diff
@@ -46,35 +48,57 @@ private ImportSpecification(string name, string version, bool isValid, TextSpan
 
         public static ImportSpecification From(SyntaxBase specificationSyntax)
         {
-            switch (specificationSyntax)
+            return specificationSyntax switch
             {
-                case StringSyntax stringSyntax when stringSyntax.TryGetLiteralValue() is { } value:
-                    var (name, version, isValid) = Parse(value);
-                    var span = isValid ? new TextSpan(stringSyntax.Span.Position + 1, name.Length) : stringSyntax.Span;
-
-                    return new ImportSpecification(name, version, isValid, span);
-
-                case SkippedTriviaSyntax trivia:
-                    return new ImportSpecification(trivia.TriviaName, trivia.TriviaName, false, trivia.Span);
+                StringSyntax stringSyntax when stringSyntax.TryGetLiteralValue() is { } value
+                    => CreateFromStringSyntax(stringSyntax, value),
+                SkippedTriviaSyntax trivia
+                    => new ImportSpecification(
+                        trivia.TriviaName,
+                        trivia.TriviaName,
+                        trivia.TriviaName,
+                        false,
+                        trivia.Span),
+                _
+                    => new ImportSpecification(
+                        LanguageConstants.ErrorName,
+                        LanguageConstants.ErrorName,
+                        LanguageConstants.ErrorName,
+                        false,
+                        specificationSyntax.Span)
+            };
+        }
 
-                default:
-                    return new ImportSpecification(LanguageConstants.ErrorName, LanguageConstants.ErrorName, false, specificationSyntax.Span);
+        public SyntaxBase ToPath()
+        {
+            if (!this.IsValid)
+            {
+                return new SkippedTriviaSyntax(this.Span, Enumerable.Empty<SyntaxBase>());
```

Comment:
Could you make the return value nullable (`SyntaxBase?`) and just return `null` here?

Reply:
I can definitely do that, however I was following the pattern that was established in the code with the TriviaSyntax. Would you be so kind to help me understand (so I can learn for the next time) the difference and explain why we favor returning null vs the current implementation in this case?

Reply:
1. Returning null is a clearer signal to the consumer that they need to consider the return result may not be valid.
1. Creating syntax in-memory is best avoided if possible - spans are used to drive diagnostics, and if the syntax was generated rather than user-authored, there's no guarantee that the span we use will make sense to the user.

> I was following the pattern that was established in the code with the TriviaSyntax

Which pattern are you referring to here - could you share a link to what you mean? That said, returning nullable types to signify a conditional return is a pattern that's **extremely** prevalent in this codebase.

Reply:
The pattern I was following was established in https://github.com/Azure/bicep/blob/main/src/Bicep.Core/Syntax/ImportSpecification.cs#L47-L63. As you can see here there I interpeted this code to make a distinction in the faliure modes for this function, it seems to use a SkippedTriviaSyntax class whose usage pattern seems to indicate that it conveys meaningful information regarding the failure that propagates to the diagnostic. Hence I chose to approach the code with care and mantain the pattern employed there with the introuction of this new function. However, based on your request and recommendation and since I accept your judgement as a manager and your engineering expertise leading to this request, I'm more than happy to refactor to return null. Thanks again for your clarification and mentoring on this topic